lib/elastic_record/as_document.rb

```diff
@@ -1,8 +1,9 @@
 module ElasticRecord
   module AsDocument
-    def as_search_document
-      doctype.mapping[:properties].each_with_object({}) do |(field, mapping), result|
-        value = value_for_elastic_search field, mapping
+    def as_search_document(mappings = doctype.mapping[:properties])
+      (mappings || []).each_with_object({}) do |(field, mapping), result|
+        nested = mappings.dig(field, :properties)
```

Comment:
Is it necessary to get the nested mapping out ahead of time (even when it does not exist?) . Only `value_for_elasticsearch` cares about this, and only once it's definitely a nested mapping.

Reply:
but the full `mappings` don't get passed into that method. I'll see what else can be done to clean things up though.

Reply:
cleaned up a little.. only thing to do is pass the mappings down, but then only the thing that needs it has to dig in.